src/wp-includes/l10n.php

```diff
@@ -702,16 +702,20 @@ function translate_nooped_plural( $nooped_plural, $count, $domain = 'default' )
  * and will be a MO object.
  *
  * @since 1.5.0
+ * @since 6.1.0 Added the `$locale` parameter.
  *
- * @global MO[] $l10n          An array of all currently loaded text domains.
- * @global MO[] $l10n_unloaded An array of all text domains that have been unloaded again.
+ * @global MO[]                   $l10n                   An array of all currently loaded text domains.
+ * @global MO[]                   $l10n_unloaded          An array of all text domains that have been unloaded again.
+ * @global WP_Textdomain_Registry $wp_textdomain_registry WordPress Textdomain Registry.
  *
  * @param string $domain Text domain. Unique identifier for retrieving translated strings.
  * @param string $mofile Path to the .mo file.
+ * @param string $locale Optional. Locale. Default is the current locale.
  * @return bool True on success, false on failure.
  */
-function load_textdomain( $domain, $mofile ) {
-	global $l10n, $l10n_unloaded;
+function load_textdomain( $domain, $mofile, $locale = null ) {
```

Comment:
```suggestion
function load_textdomain( $domain, $mofile, $locale = '' ) {
```

Since null isn't documented

Reply:
We do that elsewhere too though, e.g. in `load_default_textdomain()`

Would the alternative be `@param string|null $locale`?

Reply:
Only noted this due to PHP 8+ changes and notices like https://core.trac.wordpress.org/ticket/55967.

But since we don't pass the `null` value to any function we may be good with just leaving it as `null`.